File: submissions/CodesRN/problem1.java
import java.util.*;
class problem1
{
    public int prime(int r)
    {   int i;int sum=0;
        for(i=1;i<=r;i++)
        {
            if(r%i==0)
            sum=sum+1; 
        }
        return sum;
    }
    public static void main(String args[])
    {
        problem1 ob=new problem1();
        int sum=0;
        int sum2=0,r=0,q=0;
        Scanner sc=new Scanner(System.in);
        //System.out.println("enter a number");
        int n=sc.nextInt();
        int nn=n;
        while(n>0)
        {
            sum=sum+1;
            n=n/10;
        }
        int p=(int)Math.pow(10,(sum-1));
        for(int i=0;i<sum;i++)
        {
           //System.out.println(nn);
            if(ob.prime(nn)==2)
            sum2=sum2+1;
            r=nn%p;
            q=nn/p;
            nn=r*10+q;
        }
        if(sum2==sum)
        System.out.println("CIRCULAR PRIME");
        else
        System.out.println("NOT  A CIRCULAR PRIME");
    }
}
    

       

File: submissions/superman/problem1.cpp
#include<stdio.h>
#include<math.h>
main()
{
	int n,r,l,i=0,s=0,c=0,f=0,sum=0;
	scanf("%d",&n);
	s=n;
	while(n!=0)
	{
		n=n/10;
		c++;
	}
	n=s;
//	sum=n;
	while(sum!=s)
	{
		r=r%10;
		n=n/10;
		sum=r*pow(10,(c-1))+n;
		for(i=2;i<sum/2;i++)
		{
			if(sum%i==0)
			{
				f=1;
				break;
			}
			
		}
		if(f=1)
		{
			break;
		}
		n=sum;
	}
	if(f==1)
	{
		printf("NOT A CIRCULAR PRIME");
	}
	else
	{
		printf("CIRCULAR PRIME");
	}
}